commons/src/types/Nylas.ts

```diff
@@ -226,6 +237,29 @@ export interface ComposerProperties extends Manifest {
   show_to: boolean;
   theme: "string";
   visible: boolean;
+
+  value: Partial<Message> | void;
+  to: ContactSearchCallback;
+  from: ContactSearchCallback;
+  cc: ContactSearchCallback;
+  bcc: ContactSearchCallback;
+  send: SendCallback;
+  open: () => void;
+  close: () => void;
+  change: FetchContactsCallback | null;
+  beforeSend: (msg: Message) => Message | void;
+  afterSendSuccess: Function | null;
+  afterSendError: Function | null;
+  template: string;
+  tracking: Tracking | null;
+
+  // Attributes
+  beforeFileUpload: Function | null;
+  afterFileUploadSuccess: Function | null;
+  afterFileUploadError: Function | null;
+  uploadFile: Function | null;
+  beforeFileRemove: Function | null;
+  afterFileRemove: Function | null;
```

Comment:
Were all these just missed as types when we first built the composer, by the looks of things?

Reply:
Yep.

Reply:
Hey @femyeda, the composer's type defs are split up a bit differently right now. The types that you've redeclared here partially exist in https://github.com/nylas/components/blob/main/commons/src/types/Composer.ts. If you're going to change the types, could you also refactor the rest of them to match how we define component props / manifest props in our other components? eg. Move `ComposerProperties` into Composer.ts, and replace the other typedefs in Composer to use `ComposerProperties` instead of the `UI` and `Attributes` defs.